src/writer/_dirent.h

```diff
@@ -77,6 +77,49 @@ namespace zim
       uint16_t m_size;
     } __attribute__((packed));
 
+    struct PathTitleTinyString : TinyString {
+      PathTitleTinyString() : TinyString() {}
+      PathTitleTinyString(const std::string& path, const std::string& title)
+        : TinyString(PathTitleTinyString::concat(path, title))
+      {}
+
+      static std::string concat(const std::string& path, const std::string& title) {
+        std::string result(path.data(), path.size()+1);
+        if ( title != path ) {
+          result += title;
+        }
+        return result;
+      }
+      std::string getPath() const {
+        if (m_size == 0) {
+          return std::string();
+        }
+        return std::string(m_data);
+      }
+      std::string getTitle() const {
+        if (m_size == 0) {
+          return std::string();
+        }
+        auto title_start = std::strlen(m_data) + 1;
+        if (title_start == m_size) {
+          return std::string(m_data); // return the path as a title
+        } else {
+          return std::string(m_data+title_start, m_size-title_start);
+        }
+      }
+      std::string getStoredTitle() const {
```

Comment:
This only slightly differs from `getTitle()`, we can merge them by using a boolean argument to get(or not get) the real title. But this is just a suggestion, its fine either way.

Reply:
Done.